ow_regolith/src/RegolithSpawner.cpp

```diff
@@ -248,7 +269,25 @@ void RegolithSpawner::removeAllRegolithModels()
   }
 }
 
-void RegolithSpawner::terrainVisualModCb(const modified_terrain_diff::ConstPtr& msg)
+void RegolithSpawner::onLinkStatesMsg(const LinkStates::ConstPtr &msg) 
+{
+  auto name_it = msg->name.begin();
+  auto pose_it = msg->pose.begin();
+  while (name_it != msg->name.end() && pose_it != msg->pose.end()) {
+    if (*name_it == m_scoop_link_name) {
+      m_scoop_orientation = Quaternion(pose_it->orientation.x, 
+                                       pose_it->orientation.y,
+                                       pose_it->orientation.z,
+                                       pose_it->orientation.w);
+      return;
+    }
+    ++name_it;
+    ++pose_it;
+  }
+  ROS_WARN("Failed to find %s in link states", m_scoop_link_name.c_str());
```

Comment:
[required] Use tf::quaternionMsgToTF instead of manually assigning the field. The STL find is still optional.
```suggestion
  auto name_it = std::find(begin(msg->name), end(msg->name), m_scoop_link_name);
  if (name_it == msg->name.end())
  {
    ROS_WARN("Failed to find %s in link states", m_scoop_link_name.c_str());
    return;
  }

  auto pose_it = msg->pose.begin() + (name_it - msg->name.begin());
  tf::quaternionMsgToTF(pose_it->orientation, m_scoop_orientation);
```